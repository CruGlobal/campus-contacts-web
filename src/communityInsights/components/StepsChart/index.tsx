import { useQuery } from 'react-apollo-hooks';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { DocumentNode } from 'graphql';

import BarChart from '../BarChart';

const Wrapper = styled.div`
    height: 400px;
    position: relative;
`;

interface Props {
    query: DocumentNode;
    mapData: (data: any) => any;
    variables: any;
    label: string;
    index: string;
    nullContent: string;
}
const StepsChart = ({
    query,
    mapData,
    label,
    index,
    variables,
    nullContent,
}: Props) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Wrapper>{t('loading')}</Wrapper>;
    }

    return (
        <Wrapper>
            <BarChart
                nullContent={nullContent}
                data={mapData(data)}
                keys={[label]}
                indexBy={index}
                onFilterChanged={() => {}}
                index={0}
                filterType={'month'}
            />
        </Wrapper>
    );
};

export default StepsChart;
