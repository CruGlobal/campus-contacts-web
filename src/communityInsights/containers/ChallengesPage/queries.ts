import gql from 'graphql-tag';

const GET_IMPACT_CHALLENGES = gql`
    query impactReportStepsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stepsCount
            }
        }
    }
`;

const GET_CHALLENGES = gql`
    query communityChallenges(
        $id: ID!
        $first: Int
        $last: Int
        $after: String
        $before: String
        $sortBy: [CommunityChallengesSortEnum!]
    ) {
        community(id: $id) {
            communityChallenges(
                first: $first
                last: $last
                after: $after
                before: $before
                sortBy: $sortBy
            ) {
                nodes {
                    id
                    title
                    acceptedCount
                    completedCount
                    createdAt
                    endDate
                }
                pageInfo {
                    hasNextPage
                    startCursor
                    endCursor
                    hasPreviousPage
                }
            }
        }
    }
`;

export { GET_CHALLENGES, GET_IMPACT_CHALLENGES };
