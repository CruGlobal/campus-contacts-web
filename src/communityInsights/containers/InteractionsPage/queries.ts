import gql from 'graphql-tag';

const INTERACTIONS_TOTAL_REPORT = gql`
    query impactReportInteractionsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                interactionsCount
                interactionsReceiversCount
            }
        }
    }
`;

const INTERACTIONS_TOTAL_COMPLETED_REPORT = gql`
    query communityReportInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                interactions {
                    nodes {
                        interactionCount
                        interactionType {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const INTERACTIONS_COMPLETED_REPORT = gql`
    query communityReportDaysInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        interactionsCount
                        interactionResults {
                            nodes {
                                count
                                interactionType {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

export {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
};
