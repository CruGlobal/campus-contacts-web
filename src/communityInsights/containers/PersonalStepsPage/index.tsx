import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import { AppContext } from '../../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
} from './queries';
// GET_IMPACT_REPORT_STEPS_TAKEN INTERFACES
import { impactReportPersonalStepsCompletedCount } from './__generated__/impactReportPersonalStepsCompletedCount';
// GET_TOTAL_STEPS_COMPLETED_SUMMARY INTERFACES
import {
    communityReportStagesPersonalStepsCompleted,
    communityReportStagesPersonalStepsCompleted_community_report_stagesReport as communityReportStagesPersonalStepsCompletedEntry,
} from './__generated__/communityReportStagesPersonalStepsCompleted';
// GET_STEPS_COMPLETED_REPORT INTERFACES
import {
    communityReportDaysPersonalSteps,
    communityReportDaysPersonalSteps_community_report_daysReport as communityReportDaysPersonalRow,
    communityReportDaysPersonalSteps_community_report_daysReport_stageResults as communityReportDaysPersonalStage,
} from './__generated__/communityReportDaysPersonalSteps';
// GET_STAGES_REPORT_STEPS_ADDED INTERFACES
import {
    communityReportStagesPersonalStepsAdded,
    communityReportStagesPersonalStepsAdded_community_report_stagesReport as communityReportStagesPersonalStepsAddedRow,
} from './__generated__/communityReportStagesPersonalStepsAdded';
// GET_IMPACT_REPORT_MOVED INTERFACES
import { impactReportStageProgressionCount } from './__generated__/impactReportStageProgressionCount';
// GET_STAGES_REPORT_MEMBER_COUNT INTERFACES
import {
    communityReportStagesPersonalMemberCount,
    communityReportStagesPersonalMemberCount_community_report_stagesReport as communityReportStagesPersonalMemberCountRow,
} from './__generated__/communityReportStagesPersonalMemberCount';

const PersonalStepsPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_STEPS_TAKEN}
                text={(report: impactReportPersonalStepsCompletedCount) =>
                    t('personalSteps.taken', {
                        count:
                            report.community.impactReport
                                .personalStepsCompletedCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('personalSteps.completedTotal')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_SUMMARY}
                    variables={{ id: orgId }}
                    mapData={(
                        data: communityReportStagesPersonalStepsCompleted,
                    ) =>
                        data.community.report.stagesReport.map(
                            (
                                entry: communityReportStagesPersonalStepsCompletedEntry,
                            ) => ({
                                stage: entry.stage.name,
                                icon: entry.stage.name
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.personalStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('personalSteps.completed')}
                subtitle={t('personalSteps.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={(data: communityReportDaysPersonalSteps) =>
                        data.community.report.daysReport.map(
                            (row: communityReportDaysPersonalRow) => ({
                                ['total']: row.personalStepsCompletedCount,
                                ['stages']: row.stageResults.map(
                                    (
                                        stage: communityReportDaysPersonalStage,
                                    ) => ({
                                        name: stage.stage.name,
                                        count:
                                            stage.personalStepsCompletedCount,
                                    }),
                                ),
                                ['date']: row.date,
                            }),
                        )
                    }
                    currentDate={moment().toDate()}
                    label={t('personalSteps.legend')}
                />
            </Card>
            <Card
                title={t('personalSteps.added')}
                subtitle={t('personalSteps.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_STEPS_ADDED}
                    mapData={(data: communityReportStagesPersonalStepsAdded) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalStepsAddedRow,
                            ) => ({
                                [t(
                                    'personalSteps.label',
                                )]: row.personalStepsAddedCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('personalSteps.label')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>

            <ImpactInfo
                query={GET_IMPACT_REPORT_MOVED}
                text={(report: impactReportStageProgressionCount) =>
                    t('personalSteps.reached', {
                        count:
                            report.community.impactReport.stageProgressionCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('personalSteps.members')}
                subtitle={t('personalSteps.membersSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT_MEMBER_COUNT}
                    mapData={(data: communityReportStagesPersonalMemberCount) =>
                        data.community.report.stagesReport.map(
                            (
                                row: communityReportStagesPersonalMemberCountRow,
                            ) => ({
                                [t('members')]: row.memberCount,
                                [t('stage')]: row.stage.name.toUpperCase(),
                            }),
                        )
                    }
                    label={t('members')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default PersonalStepsPage;
