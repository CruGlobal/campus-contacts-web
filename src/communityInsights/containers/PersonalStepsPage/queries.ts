import gql from 'graphql-tag';

const GET_IMPACT_REPORT_MOVED = gql`
    query impactReportStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stageProgressionCount
            }
        }
    }
`;

const GET_IMPACT_REPORT_STEPS_TAKEN = gql`
    query impactReportPersonalStepsCompletedCount($id: ID!) {
        community(id: $id) {
            impactReport {
                personalStepsCompletedCount
            }
        }
    }
`;

const GET_STAGES_REPORT_MEMBER_COUNT = gql`
    query communityReportStagesPersonalMemberCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        memberCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT_STEPS_ADDED = gql`
    query communityReportStagesPersonalStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysPersonalSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        personalStepsCompletedCount
                        stageResults {
                            nodes {
                                personalStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_SUMMARY = gql`
    query communityReportStagesPersonalStepsCompleted(
        $id: ID!
        $period: String!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
};
