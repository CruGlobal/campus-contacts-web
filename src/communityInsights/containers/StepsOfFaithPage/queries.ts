import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReportOtherStepsCompleted($id: ID!) {
        community(id: $id) {
            impactReport {
                othersStepsCompletedCount
                othersStepsReceiversCompletedCount
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityReportStagesOthersStepsCompleted(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysOthersSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        othersStepsCompletedCount
                        stageResults {
                            nodes {
                                othersStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query communityReportStagesOthersStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReportMembersStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                membersStageProgressionCount
            }
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query communityReportStagesOthersContactCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        contactCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
