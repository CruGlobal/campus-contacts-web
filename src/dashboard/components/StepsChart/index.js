import { useQuery } from 'react-apollo-hooks';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';

import BarChart from '../BarChart';

const Wrapper = styled.div`
    height: 400px;
    position: relative;
`;

const StepsChart = ({ query, mapData, label, index, variables }) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Wrapper>{t('loading')}</Wrapper>;
    }

    return (
        <Wrapper>
            <BarChart data={mapData(data)} keys={[label]} indexBy={index} />
        </Wrapper>
    );
};

export default StepsChart;
