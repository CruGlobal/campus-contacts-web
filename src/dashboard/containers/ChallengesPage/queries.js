import gql from 'graphql-tag';

const GET_IMPACT_CHALLENGES = gql`
    query communityReport {
        communityReport {
            impactReport {
                interactionsCount
                interactionsReceiversCount
                challengesCount
            }
        }
    }
`;

const GET_CHALLENGES = gql`
    query globalCommunityChallenges($id: ID!, $first: Int, $after: String) {
        community(id: $id) {
            communityChallenges(first: $first, after: $after) {
                nodes {
                    id
                    title
                    acceptedCount
                    completedCount
                    createdAt
                    endDate
                }
                pageInfo {
                    hasNextPage
                    startCursor
                    endCursor
                    hasPreviousPage
                }
            }
        }
    }
`;

export { GET_CHALLENGES, GET_IMPACT_CHALLENGES };
