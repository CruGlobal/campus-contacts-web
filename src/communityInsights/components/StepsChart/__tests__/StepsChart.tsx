import React from 'react';
import gql from 'graphql-tag';
import moment from 'moment';
import { waitForElement } from '@testing-library/react';

import { renderWithContext } from '../../../../testUtils';
import { GET_STAGES_REPORT_STEPS_ADDED } from '../../../containers/PersonalStepsPage/queries';
import StepsChart from '..';

describe('<StepsChart />', () => {
    it('should render properly', async () => {
        const { snapshot, unmount } = renderWithContext(
            <StepsChart
                nullContent={'personalStepsAdded'}
                query={GET_STAGES_REPORT_STEPS_ADDED}
                mapData={data =>
                    data.community.report.stagesReport.nodes.map(
                        (row: any) => ({
                            ['label']: row.personalStepsAddedCount,
                            ['index']: row.stage.name.toUpperCase(),
                        }),
                    )
                }
                label={'label'}
                index={'index'}
                variables={{
                    period: '',
                    id: 1,
                    endDate: moment('2019-07-21').format(),
                }}
            />,
        );
        snapshot();
        unmount();
    });
    // This test broke with the responsive bar.
    xit('should render properly with data', async () => {
        const { snapshot, getByText } = renderWithContext(
            <StepsChart
                nullContent={'personalStepsAdded'}
                query={GET_STAGES_REPORT_STEPS_ADDED}
                mapData={data =>
                    data.community.report.stagesReport.nodes.map(
                        (row: any) => ({
                            ['label']: row.personalStepsAddedCount,
                            ['index']: row.stage.name.toUpperCase(),
                        }),
                    )
                }
                label={'label'}
                index={'index'}
                variables={{
                    period: '',
                    id: 1,
                    endDate: moment('2019-07-21').format(),
                }}
            />,
            {
                mocks: {
                    Query: () => ({
                        community: () => ({
                            report: () => ({
                                stagesReport: () => ({
                                    nodes: () => [
                                        {
                                            stepsAddedCount: 20,
                                            stage: {
                                                name: 'NO STAGE',
                                            },
                                        },
                                        {
                                            stepsAddedCount: 25,
                                            stage: {
                                                name: 'UNINTERESTED',
                                            },
                                        },
                                    ],
                                }),
                            }),
                        }),
                    }),
                },
                appContext: {
                    orgId: '1',
                },
            },
        );

        await waitForElement(() => getByText('NO STAGE'));
        snapshot();
    });
});
