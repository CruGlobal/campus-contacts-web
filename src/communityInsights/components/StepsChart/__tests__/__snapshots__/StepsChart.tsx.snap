// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsChart /> should render properly 1`] = `
.emotion-0 {
  height: 400px;
  position: relative;
}

<div>
  <div
    class="emotion-0"
  >
    Loading...
  </div>
</div>
`;

exports[`<StepsChart /> should render properly with data 1`] = `
.emotion-0 {
  height: 400px;
  position: relative;
}

.emotion-1 {
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="emotion-0"
  >
    <div>
      <div
        class="emotion-0"
      >
        <div
          style="position: relative;"
        >
          <svg
            height="400"
            role="img"
            width="848"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="transparent"
              height="400"
              width="848"
            />
            <g
              transform="translate(30,30)"
            >
              <g>
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="280"
                  y2="280"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="252"
                  y2="252"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="224"
                  y2="224"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="196"
                  y2="196"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="168"
                  y2="168"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="140"
                  y2="140"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="112"
                  y2="112"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="83"
                  y2="83"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="55"
                  y2="55"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="27"
                  y2="27"
                />
              </g>
              <g
                transform="translate(0,280)"
              >
                <g
                  style="opacity: 1;"
                  transform="translate(239,0)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="text-before-edge"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="middle"
                    transform="translate(0,20) rotate(0)"
                  >
                    NO STAGE
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(579,0)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="text-before-edge"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="middle"
                    transform="translate(0,20) rotate(0)"
                  >
                    UNINTERESTED
                  </text>
                </g>
                <line
                  style="stroke: transparent; stroke-width: 1;"
                  x1="0"
                  x2="818"
                  y1="0"
                  y2="0"
                />
              </g>
              <g
                transform="translate(0,0)"
              >
                <g
                  style="opacity: 1;"
                  transform="translate(0,280)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    0
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,252)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    10000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,224)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    20000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,196)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    30000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,168)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    40000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,140)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    50000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,112)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    60000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,83)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    70000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,55)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    80000
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,27)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    90000
                  </text>
                </g>
                <line
                  style="stroke: transparent; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="280"
                />
              </g>
              <g>
                <g
                  transform="translate(137, 163)"
                >
                  <rect
                    fill="#3CC8E6"
                    height="117"
                    rx="0"
                    ry="0"
                    stroke="#3CC8E6"
                    stroke-width="0"
                    width="204"
                  />
                </g>
                <g
                  transform="translate(477, 0)"
                >
                  <rect
                    fill="#3CC8E6"
                    height="280"
                    rx="0"
                    ry="0"
                    stroke="#3CC8E6"
                    stroke-width="0"
                    width="204"
                  />
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="emotion-1"
      />
    </div>
  </div>
</div>
`;
